test/extended/apiserverauth/api.go

```diff
@@ -0,0 +1,40 @@
+package apiserverauth
```

Comment:
@Phaow apiserverauth ? Any reason behind choosing this name ?

Reply:
Seems he wants to design sub-directories per components / features / current sub-teams :) . FYI https://github.com/kcp-dev/kcp/tree/main/test/e2e is the feature sub-dirs, not sure reference-able?

Reply:
@xingxingxia  if i understand correctly, currently our goal is to design smoke/sanity cases and not anything with respect to feature related and please correct me if i am wrong :-). So i think this should be named as kcpbasics or something like that i feel , WDYS ?

Reply:
OK, but I'm not sure of your thought. If say it is only the design of current goal, does it mean as time goes by, the design will need change after current goal gets achieved and new goal arrives? About smoke / sanity, how about using labels [Smoke] like https://docs.google.com/document/d/1k761p65J0Ig81hwZWaw73QJV6SBa7kPxbjuTEBlBbF4/edit#heading=h.irspvef076fy ? Anyway, I don't object kcpbasics as package name

Reply:
Not really, when we start automating specific cases i think it is better to have such specific naming then(like equivalent to what dev has). We are calling this apiserver but actually we have cases like version check & creation of workspaces so AFAIU name is not suitable here.

Reply:
@kasturinarra @xingxingxia Thanks for the advice! From my understanding,  design smoke/sanity cases  and design sub-directories per components/features don't have conflict, I just want to design it better for furture when kcp-qe member grows more and more like ocp for the sub components isolation maybe better. WDYT? Or any concern at here. The cases is just examples for like the case number just use to verify the framework works well and the key point is how we want to design the test suits.

Reply:
@Phaow i understand your point, but apiserverauth does not make sense to me at this point of time and it more feels like ocp to me rather than kcp, just my 2 cents, but if everyone agrees to that is how it should be then no problem for me.

Reply:
@Phaow since these are basic cases and do not have any involvement with apiserverauth team, can we keep the package name as kcpsmoke/ kcpsanity ?

Reply:
As we already has the `[Smoke]` tag, from my understanding when we execute the cases of smoke the cases could be in apiauth/workload/storage/kcpsmoke/ kcpsanity etc. Maybe kcpsmoke/ kcpsanity is not a good package name. Or could we add it as a common package like upstream, WDYT?

Reply:
From my side, I think with the help of the `[Smoke]` tag, maybe it is not mandatory to have a specific package/sub-directory for smoke. Like in the further if we plan to design our packages/sub-directories with different sub-features, what we need is to add cases in individual sub-directory and add the `[Smoke]` tag if needed. No need to consider if it is a smoke case now or if it might change as time goes by.
I think @kasturinarra's concern might be these new cases don't belong to `apiserverauth`, I think we could use something like kcpbasics in this PR, but I agree to add new cases in individual sub-directory when adding new cases (even if it is a smoke case) if it belongs to a specific feature. It could be handled easily.